docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,2165 @@
+- **Glue**: is a **single-layered** `Communication Layer`.
+- **Client**: is a **single-layered** `Communication Layer`.
+- **Service**: is a **single-layered**, overarching `Business Layer`.
+- **Shared**: is a layer-overarching, stateless, abstraction library.
+
+## Zed
+
+`Zed` serves as the backend application layer responsible for housing all business logic and persisting data.
+```
+[Organization]
+└── Zed
+    └── [Module]
+        ├── Presentation
+        │   ├── Layout
+        │   │   └── [layout-name].twig
+        │   └── [name-of-controller]
+        │       └── [name-of-action].twig
+        │    
+        ├── Communication
+        │   ├── Controller
+        │   │   ├── IndexController.php
+        │   │   ├── GatewayController.php
+        │   │   └── [Name]Controller.php
+        │   ├── [CustomDirectory]
+        │   │   ├── [ModelName]Interface.php
+        │   │   └── [ModelName].php        
+        │   ├── Exception
+        │   ├── Plugin
+        │   │   └── [ConsumerModule]
+        │   │       └── [PluginName]Plugin.php
+        │   ├── Form
+        │   │   ├── [Name]Form.php
+        │   │   └── DataProvider
+        │   │       └── [Name]FormDataProvider.php
+        │   ├── Table
+        │   │   └── [Name]Table.php                
+        │   ├── navigation.xml       
+        │   └── [Module]CommunicationFactory.php
+        │            
+        ├── Business
+        │   ├── [CustomDirectory]
+        │   │   ├── [ModelName]Interface.php
+        │   │   └── [ModelName].php        
+        │   ├── Exception
+        │   ├── Installer
+        │   │   ├── [Module]InstallerInterface.php
+        │   │   └── [Module]Installer.php        
+        │   ├── [Module]BusinessFactory.php
+        │   ├── [Module]Facade.php
+        │   └── [Module]FacadeInterface.php
+        │
+        ├── Persistence
+        │   ├── [Module]EntityManagerInterface.php
+        │   ├── [Module]EntityManager.php
+        │   ├── [Module]QueryContainer.php
+        │   ├── [Module]PersistenceFactory.php
+        │   ├── [Module]RepositoryInterface.php
+        │   ├── [Module]Repository.php
+        │   └── Propel
+        │       ├── Abstract[TableName].php
+        │       ├── AbstractSpy[TableName]Query.php
+        │       ├── Mapper
+        │       │   └── [Module|Entity]Mapper.php
+        │       └── Schema
+        │           └── [organization_name]_[domain_name].schema.xml
+        │
+        ├── Dependency
+        │   ├── Client
+        │   │   ├── [Module]To[Module]ClientBridge.php
+        │   │   └── [Module]To[Module]ClientInterface.php          
+        │   ├── Facade
+        │   │   ├── [Module]To[Module]FacadeBridge.php
+        │   │   └── [Module]To[Module]FacadeInterface.php
+        │   ├── Service
+        │   │   ├── [Module]To[Module]ServiceBridge.php
+        │   │   └── [Module]To[Module]ServiceInterface.php      
+        │   └── Plugin
```

Comment:
IMO this is mostly exception. I can remember only MoneyPlugin as an example of such dependency

Reply:
We have 69 plugin stack definition in Yves (eg: `SprykerShop\Yves\SessionCustomerValidationPageExtension\Dependency\Plugin\CustomerSessionValidatorPluginInterface`). Or maybe i did not understand your point?